OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs

```diff
@@ -159,5 +194,41 @@ void LoadMod(ModMetadata mod)
 				Game.InitializeMod(mod.Id, null);
 			});
 		}
+
+		static bool IsModInstalled(ModMetadata mod)
+		{
+			var manifest = new Manifest(mod.Id);
+			var folders = manifest.Folders.Where(f => f.Contains("^")).Select(f => f.Substring(f.IndexOf("^", StringComparison.Ordinal)));
+
+			var filesList = new List<string>();
+			foreach (var folder in folders)
+			{
+				var path = Platform.ResolvePath(folder).Replace("/", "\\");
+				if (!Directory.Exists(path))
+					continue;
+
+				var filesInDir = Directory.GetFiles(path).Select(f => f.ToLower());
+				filesList.AddRange(filesInDir);
+			}
+
+			var files = filesList.ToHashSet();
+
+			return mod.AssetInformation.TestFiles.All(f => files.Contains(Platform.ResolvePath(f).Replace("/", "\\").ToLower()));
```

Comment:
This combines both mistakes of above and breaks the detection on non-Windows systems completely. The following will work:

```charp
return mod.AssetInformation.TestFiles.All(f => files.Contains(Platform.ResolvePath(f)));
```